### PR TITLE
[travis] use ccache in LXD builds

### PR DESCRIPTION
**Pull request by Saviq**

### CODE REVIEW

**Comment by Saviq** [comment]
This seems to make it slower rather than faster... ¿?

File: tests/travis-Debug.patch [hunk 1]
```diff
@@ -1,13 +1,13 @@
 --- a/snap/snapcraft.yaml
 +++ b/snap/snapcraft.yaml
-@@ -143,9 +143,8 @@ parts:
-     - golang
```
Comment (ricab) [inline tests/travis-Debug.patch]: Is this somehow related to ccache?

Comment (Saviq) [reply tests/travis-Debug.patch]: Note this is a diff of a diff, so this isn't an actual change of the resulting file, but rather just some diff fuzz.

Comment (ricab) [reply tests/travis-Debug.patch]: Right, just the context lines changing... got to get used to diff-squared. Please don't ever go beyond that :)

**Comment by Saviq** [comment]
Some travis slowness ensued...

bors r=ricab
